### PR TITLE
Verification of Packages

### PR DESCRIPTION
**Pull request by ArquintL**
Extends the IDE to support not only the verification of single files but multiple files belonging to the same package. To this end, the verify command has been extended to 3 commands: verify currently open file, verify package belonging to currently open file, and a default verify command that executes one of the prior two depending on the settings.

### CODE REVIEW

File: client/src/VerificationService.ts [hunk 1]
```diff
@@ -130,43 +204,48 @@ export class Verifier {
...
     // only verify if it is a gobra file or a go file where the verification was manually invoked.
-    if (!fileUri.endsWith(".gobra") && !(fileUri.endsWith(".go") && event == IdeEvents.Manual)) return;
+    const hasNonGobraAndNonGoFiles = fileUris.some(fileUri => !Verifier.isGoOrGobraPath(fileUri.fsPath));
+    const hasGoFiles = fileUris.some(fileUri => fileUri.fsPath.endsWith(".go"));
+    if (hasNonGobraAndNonGoFiles) {
+      return;
+    }
+    if (hasGoFiles && event != IdeEvents.Manual) {
+      return;
+    }
```
Comment (jcp19) [inline client/src/VerificationService.ts:214]: shouldn't it fail more visibly when `hasNonGobraAndNonGoFiles` is true?

**Review by jcp19 — APPROVED** [review]
Looks good to me, but I might have missed some details because I am not super familiar with the codebase of the IDE

File: client/src/Helper.ts [hunk 2]
```diff
@@ -358,8 +382,7 @@ export class Commands {
   /**
     * Commands handled by Gobra-Server
     */
-  public static verifyGobraFile = "gobraServer/verifyGobraFile";
-  public static verifyGoFile = "gobraServer/verifyGoFile";
+  public static verify = "gobraServer/verify";
```
[the remark this answers is not in the document]
Comment (ArquintL) [reply client/src/Helper.ts:385]: On the server side there is just a single command which takes 1 or more inputs. The Gobra feature of just entering a package name is not supported in the IDE as there is imho no need for this (you open a file belonging to this package and press verify package)

File: client/src/VerificationService.ts (hunk 1 above)
Comment (jcp19) [reply client/src/VerificationService.ts]: Also, what happens if we have a `go.mod` file in the package? Does this set the flag `hasNonGobraAndNonGoFiles`?

File: client/README.md [hunk 3]
```diff
@@ -31,16 +31,27 @@ Any Gobra program can be converted to a Go program and back by running the comma
 This extension caches verification results by default.
 The cache can be flushed by running the command `Gobra: Flush ViperServer cache` from the command palette.
 
+### Verification of Packages
+A package can be split across multiple `.go` or `.gobra` files.
+To verify them together, run the command `Gobra: Verify currently opened package` while having on of these files open.
```
Comment (jcp19) [inline client/README.md:36]: typo: "having on" -> "having one"

**Review by jcp19 — APPROVED** [review]

**Pull request merged**